templates/node/src/Payload.ts.twig

```diff
@@ -0,0 +1,58 @@
+import { readFileSync } from "fs";
+import { File } from "node-fetch-native-with-agent";
+import { basename } from "path";
+
+export class Payload extends File {
+  private data: Buffer;
+  private fileName: string;
+
+  constructor(data: Buffer, name?: string) {
+    super([data], name || 'unnamed');
+    this.data = data;
+    this.fileName = name || 'unnamed';
+  }
+
+  public getData(): Buffer {
+    return this.data;
+  }
+
+  public getName(): string {
+    return this.fileName;
+  }
+
+  public toBinary(): Buffer {
+    return this.data;
+  }
+
+  public toJson(): any {
+    return JSON.parse(this.data.toString("utf-8"));
+  }
+
+  public toString(): string {
+    return this.data.toString("utf-8");
+  }
+
+  public toFile(): File {
+    return new File([this.data], this.fileName);
+  }
+
+  public static fromBinary(bytes: Buffer, name?: string): Payload {
+    return new Payload(bytes, name);
+  }
+
+  public static fromJson(object: any, name?: string): Payload {
+    const data = Buffer.from(JSON.stringify(object), "utf-8");
+    return new Payload(data, name);
+  }
+
+  public static fromString(text: string, name?: string): Payload {
+    const data = Buffer.from(text, "utf-8");
+    return new Payload(data, name);
+  }
+
+  public static fromFile(filePath: string, name?: string): Payload {
```

Comment:
I would call it just `path`, it's already in namespace of `file` since its in `fromFile`

Reply:
@christyjacob4 @loks0n, there is a lot of back-and-forth in this. Can we finalise it?